framework/src/iotbus/iotbus_uart.c

```diff
@@ -388,10 +390,27 @@ int iotbus_uart_write(iotbus_uart_context_h hnd, const char *buf, unsigned int l
 	}
 
 	fd = handle->fd;
+
+	loopback = TCLBON;
+	ret = tcloopback(fd, &loopback);
```

Comment:
This framework api is used for normal uart write operation. But, if you set loopback always, then normal uart writes will always fail. Instead, we need to set loopback only if we are performing loopback test. I suggest that you must add new framework apis like iotbus_uart_loopback_enable and iotbus_uart_loopback_disable and perform the loopback enable/ disable only in such case.